### DISCUSSION TITLE
Design: UI/UX revamp

### DISCUSSION THREAD

**Discussion by marcopavanello** [tenacityteam/tenacity-legacy#448] [Ideas]
After having some discussions with the team about the UI/UX direction of the project, it turned out that we need more feedback from the users (from you).
In the last few days @akleja and me worked on some different mockups of what we think Tenacity should look like in the future, however they are widely different and can't decide upon which one the development should go.
Although not finished yet and totally in wip, we think it's a good thing to get feedback now as our designs are 2 different directions (but similar in some pieces).

Here is the design by @akleja :
![image](https://user-images.githubusercontent.com/8861988/128686368-4da047b5-c689-4b41-aebc-96d4a5d38166.png)

And here is mine:
![untitled4](https://user-images.githubusercontent.com/8861988/128686507-74da6142-7df4-4f5a-9f3c-6ec646669f28.png)

Again though, keep in mind they are not anything near final, but it would be interesting to see what direction we should take in the future, let us know.

**Comment by n0toose** [comment]
I think it'll be a good idea if users have *at the very least* the option to move the meters to the right side of the screen, like akleja proposed. Dunno.

**Comment by OleksiySitka** [comment]
Is it about how things look, or how things work? 

If the latter, I'd suggest starting with a little more in-depth user research, to understand user profiles and needs, maybe to have some questionnaire, write some user stories, then, when it comes to discussions like this, present wireframes instead of screenshots in different styles.

If it's just about the style, then both look good :)

**Comment by poyribn** [comment]
The waveforms in the 2nd concept are way too imprecise for most uses.
And even in the 1st concept, they are still a bit smoothed-out which makes them slightly imprecise.

Of course, making things look good is important, but this shouldn't come at the cost of functionality.
If somebody wants another waveform style, it is good as an option.
But I hope that Tenacity sticks with Audacity-style "Raw" waveforms as the default as they're the most practical.

**Reply by n0toose** [reply]
The waveform isn't going too look like this in the final result, it's just the artifact of the program that Marco used to draw this.

**Reply by poyribn** [reply]
I hope so.

**Reply by akleja** [reply]
Yeah both waveforms are just an proximation in both concepts. I'd want to keep the drawing mostly as is, just with some light anti aliasing.

**Comment by poyribn** [comment]
About the loudness meters:
I prefer having them cut into flat-color zones instead of a color gradient, as this makes the status of my audio much clearer.

If I see orange in the color gradient, I'm not sure if the audio is a good loudness or too loud.

With flat-color zones this is much better: If I see yellow, my audio is a good loudness. If I see red, I must make it quieter.

**Reply by akleja** [reply]
Yeah I agree. I want the split to be at -18 dBFS, but it should be possible to adjust.
I'm not sure about how the RMS meter should handle color splitting, that's it's whole own discussion.
See https://github.com/tenacityteam/tenacity/issues/399 for more of those thoughts.
I've only drawn the current split peak/rms style meter so far, I'll add a proper peak meter to my concept in a while.

**Comment by iMartyn** [comment]
Gotta say, having the time in the top area makes more sense to me than putting it at the bottom.

**Comment by akleja** [comment]
Ok so I've worked some more on my mockup. 
* Better peak meters
* Master meters on side and tools att top.
* A new tool settings area (wip).
   This could be used for spectral tool settings/readouts and a some options I'd like to add to the pen tool,

There are still some things to figure out. The RMS meter for one. Also not sure about the waveform colors, but thought perhaps a bit more contrast might be a good thing. Haven't figured out track input assignment yet either

![mockup3](https://user-images.githubusercontent.com/49785979/128764491-8ed0af1c-f7ea-4bda-aed8-57f2080fce32.png)

**Reply by sashakoshka** [reply]
This is definitely the best one I have seen

**Reply by lanodan** [reply]
> tool option area

Does this also includes things like being able to see and switch audio sources? (which currently is quite a content-size mess)

**Comment by marcopavanello** [comment]
![untitled7](https://user-images.githubusercontent.com/8861988/128867910-287d8c1c-d2ad-4068-b05d-bf5ba8677e0d.png)

Updated and simplified a bit the structure of the meters, the top panel, improved the waveform visuals.
Here's a 3D view of the hierarchy structure:

![Screenshot from 2021-08-10 14-40-22](https://user-images.githubusercontent.com/8861988/128868388-b8804fec-8091-47b2-9cae-f81ee26e6b2d.png)

**Reply by poyribn** [reply]
I can't say which one is better.
Apart from my preference for the much more functional "raw" audio waveform visuals of @akleja's design (for me this is a must), both now look equally good to me.

**Reply by peepo5** [reply]
This looks really clean and I would love to use something like this.

**Comment by mc776** [comment]
Could that waveform colour be made customizable?

**Reply by marcopavanello** [reply]
In both cases the answer is yes :) in my design for example there are 2 colors defining the overall look:
- Primary color for the buttons and panels
- Secondary color for the waveform and meters

**Reply by iMartyn** [reply]
What's the third meter zone going to look like, even darker shade of the same color? I personally really think that meters need to be seperatable from the scheme so that you can have that lovely look and then meters green,yellow,red.  Meters that aren't those three colours are "unusual".

**Reply by poyribn** [reply]
Yes:
Red-Yellow-Green it should be by default.

Of course, it should be possible to change these colors, so that colorblind users can also use Tenacity without compromises.
But these classic colors that are well-known from Traffic Lights are a great default.

---

Speaking of colorblindness: If done well, the 3 colors would naturally have different values, like this:

- Yellow is the brightest
- Red and Green are darker than yellow

In that case, the option to change the colors will have a lower priority as it's already much easier to differentiate between the default colors for colorblind people as well as other people.

Reguardless, this theming possibility should still be made eventually.

**Reply by mc776** [reply]
I'd really rather that something like the OP be the default - I've always found that red-yellow-green progression really distracting in the Winamp days and it never really seemed to add anything that the two shades of blue/purple here aren't doing better.

Even if the values were different enough so that they'd be visible, it sends an *initial* message that accsesibility is an opt-in afterthought in this case.

Really, the flatter blue (or whatever) and grey is so much better looking and already accessible *as a default* that I don't see any reason why Tenacity should depart from that.

**Reply by iMartyn** [reply]
I appreciate what you're saying about looks but meters exist for a very good reason, and the colours have meaning. I personally strongly disagree that the default should be flat single colour, that's to me the very definition of form over function.  I shouldn't have to change settings to make the fundemental purpose of meters useful.

To say that a flat colour is "accessible by default" is like saying you don't need to describe your images, because they won't be seen, so it's more accessible to blind people.  Removing the _important_ functionality of the decibel zones because colourblind people can't discern them in the default colours is a bad argument.

Either make them three easy to discern for colourblind people colours and say to hell with the theme colours, or make it an easy-to-find option for colourblind people to select three colours that they can discern (point in case: there are multiple forms of colourblindness, a "default" is never going to be good for each).

The fact that you think that two shades of the same colour for a 3-zone meter is "better" than red-yellow-green and referencing winamp implies you're thinking of this from a media player point of view, not a recording studio point of view.  I would posit that for tenacity, the latter is more the target audience.  

(edited because my fingers typed audacity when my brain meant tenacity)

**Reply by mc776** [reply]
I get what you're getting at about form over function, it's just that the RYG stuff actually kinda hurts my eyes to look at so it *is* functional in that sense.

Anyway, flat versus RYG is a false dichotomy when the default could be blue on the outside, yellow in the middle and some kind of dark red on the inside and it would look a lot clearer.

**Reply by mc776** [reply]
> blue on the outside, yellow in the middle and some kind of dark red on the inside

I just realized I was unconsciously thinking about the Audacity logo colours here...

**Comment by ghost** [comment]
Great stuff. Some points

- Is this for the default theme? Tenacity will be themeable, right? If yes, then no much worries here.
-But, for the default version, I like @akleja one.
- Ardour has the Draw "flat" buttons and Draw "boxy" buttons options. I think those are nice options. With those options, I could go with @marcopavanello theme and flat and boxy options on. :-)

**Comment by akleja** [comment]
I just want to add some of my perspectives on this whole discussion

While it might be reasonable to decide on a overall goal for the UI, I am of the opinion that these are just drafts, and whichever path we chose, much will still change along the way. They might help making all small changes lead in one direction, but they should not be considered final in any way. At least that's my preference.

We need to figure out a lot of things, and I think detailed wireframes of specific components is a worthwhile focus moving forward. Trying to solve everything at once won't really work out I think. Also there are a lot of things besides the main window that needs to be reworked.

There already are a few discussions going on regarding some components, and it might be better to not have parallel discussions. I suggest using the already existing issues, or opening new ones for things not yet discussed.

Some UI related issues (might've missed some):
Meter design:https://github.com/tenacityteam/tenacity/issues/399
Device toolbar: https://github.com/tenacityteam/tenacity/issues/435#event-5134124364
Spectrograms: https://github.com/tenacityteam/tenacity/issues/237
Mixer interface: https://github.com/tenacityteam/tenacity/issues/237

There's also some prior discussions:
Vertical docks: https://github.com/tenacityteam/tenacity/discussions/398
Effects interface: https://github.com/tenacityteam/tenacity/discussions/388

**Comment by JPcode05** [comment]
I would say that we wait until maybe Version 2.0 of Tenacity to do this. We're already making an alternative to Audacity, and we should keep the UI like it was in Audacity before making a change to it. We don't want to throw those off coming from Audacity, of course.

**Reply by peepo5** [reply]
Unless we make it worse why would it throw off people coming from audacity?

**Reply by poyribn** [reply]
@peepopoggers exactly. we should keep the u-i as-is, because we don't want to make it worse.

we first need a working version with the Existing u-i before making any changes. only by this way can we actually know how to make it better.

**Reply by nawordar** [reply]
@Sosasees How it could help us understand how to make better UI? Tenacity does not track how people are using it, so there is no data to compare.

**Reply by Semisol** [reply]
@nawordar Community feedback.

**Reply by poyribn** [reply]
@nawordar this will improve the program in general, because:

the first priority should be to get the program
- Rebranded from audacity to tenacity.
- actually Work.

Only once we are sure that this is done right, should we redesign the u-i (even if we already know the changes we will make), since doing Three major tasks (as suppose as just two) at once
makes it too much harder than needed
to develop the program.

**Reply by nawordar** [reply]
Never mind, this makes sense. We should make a first stable release with old UI before changing it

**Reply by peepo5** [reply]
Agreed but I think we have the great opportunity to improve the UI and after stable releases, there is no reason to not get to work.

**Reply by akleja** [reply]
I thinks some changes should be included in the first release, namely making sure the the themes apply correctly. Right now the drop downs doesn't conform to the program theme, and on some platforms doesn't match the system dark theme preference. Also the scaling of the drop downs are buggy and they get clipped on many systems.
The themeability of  the drop downs is a long standing issue with wxWidgets and not something we can fix without changing the UI.

To resolve this i think a few  changes are warranted (and some ongoing)
1. Remake the device selection interface. See:
  https://github.com/tenacityteam/tenacity/issues/435
  https://github.com/tenacityteam/tenacity/pull/543
2. Move the project sample rate into settings, it shouldn't live in a toolbar anyway
3. Change controls used in selection toolbar, perhaps by changing the selection toolbar to use a combination of labels or icon in combination with a popup menu instead of drop downs.

Additional major design changes I think I agree should wait until after first release.

**Comment by Reaper10** [comment]
Just an idea, you could take a page out of waveform daw book
![Waveform-Free-DAW-1024x576](https://user-images.githubusercontent.com/8137561/156441650-1c2159db-e239-4a81-ad10-4562b2c8fe30.jpg)
